code/controllers/subsystem/timer.dm

```diff
@@ -79,28 +83,29 @@ SUBSYSTEM_DEF(timer)
 
 	var/static/list/spent = list()
 	var/static/datum/timedevent/timer
-	var/static/datum/timedevent/head
+	var/static/current_offset = 1
```

Comment:
And here is the change that fixed the timer bug for anybody curious. 

By saving the head during pauses, if the head got deleted some other way, this would resume, head would still be the now qdeleted head, and the if (timer == head) checks below would never trigger, looping until it ran out of tick time, then resuming.

shifting buckets fixed only because doing so also forces head to get updated.